img_proof/ipa_gce.py

```diff
@@ -287,50 +498,85 @@ def _validate_region(self):
     def _get_instance_state(self):
         """Attempt to retrieve the state of the instance."""
         instance = self._get_instance()
-        return instance.state
+        return instance['status']
 
     def _is_instance_running(self):
         """Return True if instance is in running state."""
-        return self._get_instance_state() == 'running'
+        return self._get_instance_state() == 'RUNNING'
 
     def _set_instance_ip(self):
         """Retrieve and set the instance ip address."""
         instance = self._get_instance()
 
-        if instance.public_ips:
-            self.instance_ip = instance.public_ips[0]
-        elif instance.private_ips:
-            self.instance_ip = instance.private_ips[0]
-        else:
-            raise GCECloudException(
-                'IP address for instance: %s cannot be found.'
-                % self.running_instance_id
-            )
+        interface = instance['networkInterfaces'][0]
+        try:
+            self.instance_ip = interface['accessConfigs'][0]['natIP']
+        except (KeyError, IndexError):
+            try:
+                self.instance_ip = interface['networkIP']
+            except KeyError:
+                raise GCECloudException(
+                    'IP address for instance: %s cannot be found.'
+                    % self.running_instance_id
+                )
 
     def _start_instance(self):
         """Start the instance."""
-        instance = self._get_instance()
-        self.compute_driver.ex_start_node(instance)
-        self.compute_driver.wait_until_running(
-            [instance],
+        self.compute_driver.instances().start(
+            project=self.service_account_project,
+            zone=self.region,
+            instance=self.running_instance_id
+        ).execute()
+
+        self._wait_on_instance(
+            'RUNNING',
             timeout=self.timeout
         )
 
     def _stop_instance(self):
         """Stop the instance."""
-        instance = self._get_instance()
-        self.compute_driver.ex_stop_node(instance)
-        self._wait_on_instance('stopped', timeout=self.timeout)
+        self.compute_driver.instances().stop(
+            project=self.service_account_project,
+            zone=self.region,
+            instance=self.running_instance_id
+        ).execute()
+
+        self._wait_on_instance(
+            'TERMINATED',
+            timeout=self.timeout
+        )
```

Comment:
Please add a comment here that the terminology is inconsistent as received from the framework. A stopped instance has "terminated" status, not "stopped"

Reply:
Well the term is not inconsistent with GCE documentation. It's just inconsistent with other cloud frameworks. Will add a note and possibly a link to GCE docs on instance state.